packages/material-ui/src/Slider/Slider.js

```diff
@@ -505,57 +504,54 @@ const Slider = React.forwardRef(function Slider(props, ref) {
     axis += '-reverse';
   }
 
-  const getFingerNewValue = React.useCallback(
```

Comment:
This block diff is about removing the useless `React.useCallback`.

Reply:
Is it there as a performance optimization? Now `getFingerNewValue` will change on every render of the `Slider`, which forces all child components to also rerender.

Reply:
> which forces all child components to also rerender.

@embeddedt How did you conclude to this supposed fact, do you have a reproduction? I couldn't observe this behavior. Looking at the source, I highly doubt it's true.

Reply:
The [React documentation](https://reactjs.org/docs/hooks-reference.html#usecallback) mentions this and I have also seen other articles talk about it.

> This is useful when passing callbacks to optimized child components that rely on reference equality to prevent unnecessary renders (e.g. `shouldComponentUpdate`).

It's possible that the effect on performance is negligible in this case, but I think avoiding unnecessary re-renders as much as possible is important in a React application, particularly on weaker devices.

Reply:
Yeah, ok, then we are good, there is no rerendering.

Reply:
> Now getFingerNewValue will change on every render of the Slider, which forces all child components to also rerender.

@embeddedt This is true, though `React.useCallback` introduces some overhead on its own.
Check out [this great blog post](https://kentcdodds.com/blog/usememo-and-usecallback) from @kentcdodds.

Reply:
Thanks for linking that blog post; I didn't consider the performance implications of `useCallback`'s overhead!